katsdpdata/prod_handler.py

```diff
@@ -406,6 +408,17 @@ def metadata_transfer_complete(self):
         met = mh.set_product_received(met)
         met_bucket = self.get_bucket_stats()
         mh.add_bucket_stats(met, met_bucket)
+        self.solr = pysolr.Solr(self.solr_url)
+        for product_type in ['MeerKATVisibilityProduct', 'MeerKATFlagProduct']:
+            query = 'CaptureBlockId:{} AND CAS.ProductTypeName:{}'.format(met['id'],product_type)
+            query_dict = {"q": query}
+            results = self.solr.search(**query_dict)
+            if results.hits == 1:
```

Comment:
What happens if there are no results? Does this code ever get run again?

If this code never gets run again and some vis and flag products are made later, then they will never have the ProposalId set?

Reply:
After re-examining the code, I discovered the following.
If  both the  vis  and the flag products dir are empty, and there is no complete token in the rdb directory, the product will be marked as pruned . However, if the complete token is present in the RDB directory and there are no vis and flags directories, the code will move the rdb files to a failed directory.

In other words, this code will be executed if both the vis and flags products are present.